src/components/Transaction.vue

```diff
@@ -1,39 +1,152 @@
 <template>
-  <div v-if="transactions => 1" :class="`transaction ${border ? 'border' : ''}`">
-    <div class="amount">
-      <font-awesome-icon :class="`icon ${origin.toLowerCase()}`" :icon="arrowIcon" />
-      <span :class="`number ${origin.toLowerCase()}`">{{ amount }}</span>
-      <span class="wit">{{ currency }}</span>
+  <FrameOutside @click="hideDetails" @focus="hideDetails">
+    <div v-if="transactions => 1" :class="`${border ? 'border' : ''}`">
+      <div class="transaction" @click="showDetails = !showDetails">
+        <img class="icon" :src="arrowIcon" alt="" />
+        <div class="amount">
+          <span :class="`number ${origin.toLowerCase()}`">{{ amount }}</span>
+          <span class="wit">{{ currency }}</span>
+        </div>
+        <div v-show="transactionType === 'value_transfer'" class="address-container">
+          <p class="origin">{{ origin }}</p>
+          <p class="address">{{ address }}</p>
+        </div>
+        <div v-show="transactionType === 'data_request'" class="address-container">
+          <p class="address">Data request</p>
+        </div>
+        <div class="">
+          <p class="date">{{ timeAgo }}</p>
+        </div>
+      </div>
+      <div v-if="showDetails">
```

Comment:
I'd move the details to its own component for the sake of simplicity and testing